contrib/nydusify/pkg/cache/cache.go

```diff
@@ -443,14 +455,22 @@ func (cache *Cache) Import(ctx context.Context) error {
 		return errors.Wrap(err, "Unmarshal cache manifest")
 	}
 
-	// Discard the cache mismatched version
+	// Discard the cache if mismatched version
 	if manifest.Annotations[utils.ManifestNydusCache] != cache.opt.Version {
 		return fmt.Errorf(
 			"unmatched cache image version %s, required to be %s",
 			manifest.Annotations[utils.ManifestNydusCache], cache.opt.Version,
 		)
 	}
 
+	// Discard the cache if mismatched RAFS FsVersion
+	if manifest.Annotations[utils.LayerAnnotationNydusFsVersion] != cache.opt.FsVersion {
```

Comment:
For `manifest.Annotations[utils.LayerAnnotationNydusFsVersion] == ""` (e.g. old cache image), we should consider it as `5`.